tb_plugin/tensorboard_plugin_torch_profiler/profiler/module_parser.py

```diff
@@ -109,30 +110,34 @@ def __init__(self):
         self.op_list_groupby_name_input = []  # For Operator-view.
         self.kernel_list_groupby_name_op = {}  # For Kernel-view.
 
-    def _build_tree(self, host_node_list):
+    # host_node_list: list of OperatorNode and ProfilerStepNode.
+    # rt_list: list of RuntimeNode with external_id=0.
+    def _build_tree(self, host_node_list, rt_list):
 
-        def build_tree_relationship(host_node_list):
+        def build_tree_relationship(host_node_list, rt_list):
             node_stack = []
             root_node = OperatorNode()
             root_node.start_time = -sys.maxsize - 1
             root_node.end_time = sys.maxsize
+            root_node.runtimes = rt_list  # Give the list of RuntimeNode with external_id=0 to root node.
             node_stack.append(root_node)
             for node in host_node_list:
                 while True:  # break loop when the node is inserted.
                     tail_node = node_stack[-1]
                     if node.start_time < tail_node.end_time:
                         if node.end_time <= tail_node.end_time:
-                            if tail_node.type != EventTypes.RUNTIME:
-                                if (node.type == EventTypes.RUNTIME and
-                                        node.external_id != 0 and
-                                        tail_node.external_id != node.external_id):
-                                    logger.warning("Operator contains Runtime but with different external id!")
-                                tail_node.children.append(node)
-                                node_stack.append(node)
-                            else:
-                                logger.error("Error in input data: runtime range should not have children!")
+                            if (node.type == EventTypes.RUNTIME and
```

Comment:
this node.type == EventTypes.RUNTIME will always be false, right?

Reply:
Yes. Removed it.